source/materials/MaterialsList.cc

```diff
@@ -15,55 +15,85 @@
 #include <G4Element.hh>
 #include <G4NistManager.hh>
 
+#include <vector>
+#include <utility> 
+
 using namespace nexus;
 using namespace CLHEP;
 
-
 namespace materials {
-  G4Material* GXe(G4double pressure, G4double temperature)
-  {
-    G4Material* mat = GXe_bydensity(GXeDensity(pressure),
-                                    temperature, pressure);
+    // Calculation of Compressibilty factor from https://next.ific.uv.es/DocDB/0011/001183/001/GasDensity_2021-04-23.pdf
+    G4double CompressibilityFactor(G4double pressure, G4double temperature) {
+        double T_r = temperature/(289.733*kelvin);
+        double P_r = pressure/ (58.420*bar);
+        double B0 = 0.083 - (0.422 / pow(T_r, 1.6));   //Second virial coefficient    
+        double z = 1 + B0*(P_r/T_r);  //Compressibility factor
+        return z;       
+    }
+
+    // Function to calculate gas density based on Xe isotopic composition
+    G4double CalculateGasDensityFromIsotopicComposition(G4double pressure, G4double temperature,
+     const std::vector<std::pair<int, double>>& isotopicComposition)   //pressure in bar , temperature in K
+      {
+        const double R = 8.314; // Ideal gas constant in J/(mol·K)
+        double average_molar_mass = 0.0; // in g/mol
+
+        for (const auto& iso : isotopicComposition) {
+            int mass_number = iso.first;
+            double percentage = iso.second/100;
+            double molar_mass = XenonMassPerMole(mass_number) / (g / mole);
+            average_molar_mass += molar_mass * percentage;
+        }
 
+        average_molar_mass /= 1000.0; //convert in kg/mol
```

Comment:
Instead of dividing by 1000 to obtain kg/mol, why don't you directly return the molar mass in kg/mol in line 44 like this:

`double molar_mass = XenonMassPerMole(mass_number) / (kg / mole);`

? Am I missing something?